src/rust/engine/src/externs/target.rs

```diff
@@ -51,6 +51,11 @@ impl Field {
     address: PyRef<Address>,
     py: Python,
   ) -> PyResult<Self> {
+    // NB: The deprecation check relies on the presence of NoFieldValue to detect if
```

Comment:
TBH I'm not sure why we do this coercion at all. Presumably we must be able to handle NoFieldValue in the `cls_none_is_valid_value` case.

Reply:
@stuhood can you shed light on why we do this at all?